garage/tf/envs/base.py

```diff
@@ -1,68 +1,54 @@
-from cached_property import cached_property
 from gym.spaces import Box as GymBox
 from gym.spaces import Discrete as GymDiscrete
 from gym.spaces import Tuple as GymTuple
 
-from garage.envs import EnvSpec
-from garage.envs import ProxyEnv
+from garage.envs import GarageEnv
 from garage.tf.spaces import Box
 from garage.tf.spaces import Discrete
 from garage.tf.spaces import Product
 
 
-def to_tf_space(space):
-    if isinstance(space, GymBox):
-        return Box(low=space.low, high=space.high)
-    elif isinstance(space, GymDiscrete):
-        return Discrete(space.n)
-    elif isinstance(space, GymTuple):
-        return Product(list(map(to_tf_space, space.spaces)))
-    else:
-        raise NotImplementedError
+class TfEnv(GarageEnv):
+    """
+    Returns a TensorFlow wrapper class for gym.Env.
 
+    Args:
+        env (gym.Env): the env that will be wrapped
+    """
 
-class WrappedCls(object):
-    def __init__(self, cls, env_cls, extra_kwargs):
-        self.cls = cls
-        self.env_cls = env_cls
-        self.extra_kwargs = extra_kwargs
-
-    def __call__(self, *args, **kwargs):
-        return self.cls(
-            self.env_cls(*args, **dict(self.extra_kwargs, **kwargs)))
-
-
-class TfEnv(ProxyEnv):
-    @cached_property
-    def observation_space(self):
-        return to_tf_space(self.wrapped_env.observation_space)
+    def __init__(self, env):
+        super().__init__(env)
+        self.action_space = self._to_garage_space(env.action_space)
+        self.observation_space = self._to_garage_space(env.observation_space)
 
-    @cached_property
-    def action_space(self):
-        return to_tf_space(self.wrapped_env.action_space)
-
-    @cached_property
-    def spec(self):
-        return EnvSpec(
-            observation_space=self.observation_space,
-            action_space=self.action_space,
-        )
-
-    @property
-    def vectorized(self):
-        return getattr(self.wrapped_env, "vectorized", False)
+    def reset(self, **kwargs):
+        return self.env.reset(**kwargs)
 
-    def vec_env_executor(self, n_envs, max_path_length):
-        return VecTfEnv(
-            self.wrapped_env.vec_env_executor(
-                n_envs=n_envs, max_path_length=max_path_length))
+    def step(self, action):
+        return self.env.step(action)
 
     @classmethod
     def wrap(cls, env_cls, **extra_kwargs):
         # Use a class wrapper rather than a lambda method for smoother
         # serialization
         return WrappedCls(cls, env_cls, extra_kwargs)
 
+    def _to_garage_space(self, space):
```

Comment:
@overrides

Reply:
```
  File "/Users/jonathon/Documents/garage/garage/garage/tf/envs/base.py", line 31, in TfEnv
    @overrides
TypeError: 'module' object is not callable
```

Reply:
Can you show me how did you import it? It might just be a small error of importing.